src/back-end/lib/resources/affiliation.ts

```diff
@@ -131,40 +124,38 @@ const resource: Resource = {
 
   update(connection) {
     return {
-      async parseRequestBody(request) {
+      async parseRequestBody(request): Promise<null> {
         return null;
       },
-      async validateRequestBody(request) {
-        const validatedAffiliationId = await validateAffiliationId(connection, request.params.id);
-        if (validatedAffiliationId.tag === 'invalid') {
+      async validateRequestBody(request): Promise<Validation<ValidatedUpdateRequestBody, UpdateValidationErrors>> {
+        const validatedAffiliation = await validateAffiliationId(connection, request.params.id);
+        if (isInvalid(validatedAffiliation)) {
           return invalid({
-            affiliation: getInvalidValue(validatedAffiliationId, undefined)
+            affiliation: getInvalidValue(validatedAffiliation, undefined)
           });
         }
-        const existingAffiliation = validatedAffiliationId.value;
+        const existingAffiliation = validatedAffiliation.value;
         if (existingAffiliation.membershipStatus === MembershipStatus.Pending) {
           if (!permissions.updateAffiliation(connection, request.session, existingAffiliation.organization.id)) {
             return invalid({
               permissions: [permissions.ERROR_MESSAGE]
             });
           }
-          return valid({
-            affiliation: existingAffiliation.id
-          });
+          return valid(existingAffiliation.id);
         }
         return invalid({
-          membershipType: ['Membership is not pending.']
+          affiliation: ['Membership is not pending.']
         });
       },
-      async respond(request) {
-        const respond = (code: number, body: Affiliation | UpdateValidationErrors | string[]) => basicResponse(code, request.session, makeJsonResponseBody(body));
+      async respond(request): Promise<Response<JsonResponseBody<Affiliation | UpdateValidationErrors>, Session>> {
```

Comment:
This type signature is good because the `Resource` type only types the response body as one of `SupportedResponseBodies` which is very vague.